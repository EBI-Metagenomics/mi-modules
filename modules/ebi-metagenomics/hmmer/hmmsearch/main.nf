
process HMMER_HMMSEARCH {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/hmmer:3.4--hdbdd923_1' :
        'biocontainers/hmmer:3.4--hdbdd923_1' }"

    input:
    tuple val(meta), path(fasta)
    path hmm_db

    output:
    tuple val(meta), path("*.tbl"), emit: tbl
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    def hmm_file = "" 
    if (!hmm_db instanceof List) {
        hmm_file = hmm_db
    }
    else{
        for(iter_file in hmm_db) {
            if (iter_file.getExtension() == 'hmm'){
                hmm_file = iter_file
                break
            }
        }
    }

    """
    hmmsearch \\
        ${args} \\
        --cpu ${task.cpus} \\
        --domtblout ${prefix}_hmmsearch.tbl \\
        -o /dev/null \\
        ${hmm_file} \\
        ${fasta}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        hmmer: \$(hmmsearch -h | grep 'HMMER' | cut -d' ' -f3)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    touch ${prefix}_hmmsearch.tbl

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        hmmer: \$(hmmsearch -h | grep 'HMMER' | cut -d' ' -f3)
    END_VERSIONS
    """
}
