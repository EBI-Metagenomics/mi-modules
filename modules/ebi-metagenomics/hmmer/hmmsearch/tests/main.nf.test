
nextflow_process {

    name "Test Process HMMER_HMMSEARCH"
    script "../main.nf"
    process "HMMER_HMMSEARCH"
    config "./nextflow.config"


    tag "modules"
    tag "modules_nfcore"
    tag "hmmer"
    tag "hmmer/hmmsearch"

    test("hmmsearch") {

        when {
            process {
                """                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/test.fasta", checkIfExists: true)
                    ]
                input[1] = file("${moduleDir}/tests/seqdb/test1.hmm*", type: 'dir', checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
                // See https://nf-co.re/docs/contributing/tutorials/nf-test_assertions for more information and examples.
            )
        }

    }

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used but keep the " - stub" suffix.
    test("sarscov2 - bam - stub") {

        options "-stub"

        when {
            process {
                """
                // TODO nf-core: define inputs of the process here. Example:
                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['sarscov2']['illumina']['test_paired_end_bam'], checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
            )
        }

    }

}
