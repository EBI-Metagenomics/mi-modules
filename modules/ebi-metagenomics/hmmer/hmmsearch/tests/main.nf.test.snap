{
    "hmmsearch- stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_hmmsearch.tbl:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,37393b1da5a14113d3290ab8b3b4c40f"
                ],
                "tbl": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_hmmsearch.tbl:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,37393b1da5a14113d3290ab8b3b4c40f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.3"
        },
        "timestamp": "2024-06-13T12:28:01.358951221"
    },
    "hmmsearch": {
        "content": [
            true
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.3"
        },
        "timestamp": "2024-06-13T12:25:33.775461342"
    }
}